tools/build_defs/framework.bzl

```diff
@@ -680,37 +677,25 @@ def _define_inputs(attrs):
     # but filter out repeating directories
     ext_build_dirs = uniq_list_keep_order(ext_build_dirs)
 
-    tools_roots = []
-    tools_files = []
     input_files = []
-    for tool in attrs.tools_deps:
-        tool_root = detect_root(tool)
-        tools_roots.append(tool_root)
-        for file_list in tool.files.to_list():
-            tools_files += _list(file_list)
-
-    for tool in attrs.additional_tools:
-        for file_list in tool.files.to_list():
-            tools_files += _list(file_list)
-
     for input in attrs.additional_inputs:
         for file_list in input.files.to_list():
-            input_files += _list(file_list)
+            input_files += list(file_list)
 
-    # These variables are needed for correct C/C++ providers constraction,
+    tools_files, manifests_from_tools = ctx.resolve_tools(tools = ctx.attr.tools_deps + attrs.additional_tools)
```

Comment:
is `manifests_from_tools` intentionally unused here?

Reply:
It is at the moment but probably should be plumbed through InputFiles to the point at which a `ctx.actions` call using `tools_files` is made and passed in there. I don't actually know what the `manifests_from_tools` argument does - its marked as experimental on the API in the bazel docs.

Reply:
Well, can you prefix it with `_` if it's not being used and add a comment for some of this context?

Reply:
I've now updated the code to propagate it to the `run_shell` call.